core/src/test/scala/unit/kafka/cluster/PartitionTest.scala

```diff
@@ -118,14 +113,11 @@ class PartitionTest {
     ), leaderEpoch = 5)
     assertEquals(4, log.logEndOffset)
 
-    val partition = new Partition(topicPartition.topic, topicPartition.partition, time, replicaManager)
-    assertTrue("Expected makeLeader to succeed",
-      partition.makeLeader(controllerId, new LeaderAndIsrRequest.PartitionState(controllerEpoch, leader, leaderEpoch,
-        isr, 1, replicas, true), 0))
-
+    val partition = setupPartitionWithMocks(leaderEpoch = leaderEpoch, isLeader = true)
```

Comment:
Hmm, I am not sure how this test works now. setupPartitionWithMocks() creates a Log that's different from the one created in line 105. So not sure how they have the same log end offset. Also, do we need both Log?

Reply:
Hmm yeah, this is strange. I think it works because even though they are separate `Log` instances, they use the same directory and log files. Let me try to fix.